Answers.md

```diff
@@ -2,10 +2,20 @@
 
 1.  What is React JS and what problems does it try and solve?
 
+React is a javascript library for building user interfaces. React was built to manage large, scalable apps that can handle large quantities of data without bogging down the actual DOM and/or webpage. It helps so you dont need to constantly refresh the whole page and you can refresh the section which is only needing to be updated. The best example is a tweet on twitter of post/wall on facebook. Before the whole page would refresh and so save on data/bandwidth server costs probobly and for the data of the user now only the section which is being updated is being refreshed
+
 1.  What does it mean to _think_ in react?
 
+## Thinking in react refers to breaking a project down into a logical react workflow. You start with a mock and you break that down into a hierarchy of components. You then build a static version of that (without state). You then identify the minimal set of mutable state that your app needs and figure out where your state should live. Finally you add your data flow.
+
 1.  Briefly describe some of the differences between a Class/Stateful component and a Functional/Presentational component.
 
+## A functional component is a regular pure JavaScript function that accepts props as its argument, and returns some JSX whereas a class based component is in fact a JavaScript class. It extends React.Component, and its only required method is render().
+
 1.  Describe state.
 
+## State is one of the two types of data in react the other one being props. It is essentially an object that determines how that component renders & behaves. It is created in the component and is changeable by using the setState() method.

```

Comment:
One big thing is that state can be updated on a component, without the whole component being updated, the component can keep living on when its state updates. But if a component gets new props, then the whole component re-renders and everything resets.